File: AspNetMvcUnleashed/0672329980_source/Chapter08/CS/MvcApplication1/MvcApplication1/Models/MovieRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace MvcApplication1.Models
{
    public class MovieRepository : IMovieRepository
    {
        private MoviesDBEntities _entities = new MoviesDBEntities();

        public IEnumerable<Movie> ListMovies()
        {
            return _entities.MovieSet.ToList();
        }

        public void CreateMovie(Movie movieToCreate)
        {
            _entities.AddToMovieSet(movieToCreate);
            _entities.SaveChanges();
        }
    }

    public interface IMovieRepository
    {
        IEnumerable<Movie> ListMovies();
        void CreateMovie(Movie movieToCreate);
    }
}

File: AspNetMvcUnleashed/0672329980_source/Chapter08/VB/MvcApplication1/MvcApplication1/Models/MovieRepository.vb
Public Class MovieRepository
    Implements IMovieRepository

    Private _entities As New MoviesDBEntities()

    Public Function ListMovies() As IEnumerable(Of Movie) Implements IMovieRepository.ListMovies
        Return _entities.MovieSet.ToList()
    End Function

    Public Sub CreateMovie(ByVal movieToCreate As Movie) Implements IMovieRepository.CreateMovie
        _entities.AddToMovieSet(movieToCreate)
        _entities.SaveChanges()
    End Sub
End Class

Public Interface IMovieRepository
    Function ListMovies() As IEnumerable(Of Movie)
    Sub CreateMovie(ByVal movieToCreate As Movie)
End Interface
